src/mono/wasm/runtime/run.ts

```diff
@@ -123,7 +118,7 @@ function set_exit_code_and_quit_now(exit_code: number, reason?: any): void {
     }
     logErrorOnExit(exit_code, reason);
     appendElementOnExit(exit_code);
-    if (exit_code !== 0 || !ENVIRONMENT_IS_WEB) {
+    if (exit_code !== 0 && !ENVIRONMENT_IS_WEB) {
```

Comment:
`quit` will only halt the process on NodeJS or V8, not in the browser.
This change effectively swallows the exception on the web.
But on  the web, we rely on exception of type `ExitStatus` to always propagate to the very top and "stop" the pending call chain.

Since we could have many call chains in progress, scheduled by `setTimeout()` we could possibly see multiple calls to this method.

We need to think this thru more.

Reply:
We end up throwing the same error multiple times,  logging, and emitting `WASM EXIT` multiple times. But I don't understand much about this, so all yours :)

Reply:
"same error" is it the same instance ?
Did we caused that by adding the unhandled exception handler on the web ? We should filter `ExitStatus` there.

Reply:
Yeah, the unhandled exception handler was also getting the error, so some other location was throwing the error too.